### PR TITLE
Update astyle to 3.4.12

### PR DESCRIPTION
**Pull request by metalefty**
as astyle 3.1 is no longer available in the latest FreeBSD.

Also, astyle upstream repository has been moved to GitLab. Switched git to obtain astyle source. 
https://gitlab.com/saalen/astyle

### CODE REVIEW

**Comment by metalefty** [comment]
There is a little bit difference in astyle result between 3.1 and 3.4. Commit 47d4ab1339b74089f33ae4a581b5f54d316a0a1f shows the difference. I'm not sure how to configure astyle to get the same result.

File: scripts/run_astyle.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 # Note: the script must be run from the root directory of the xrdp repository
 
 INSTALL_ROOT=~/astyle.local
-ASTYLE_FROM_XRDP=$INSTALL_ROOT/3.1/usr/bin/astyle
+ASTYLE_FROM_XRDP=$INSTALL_ROOT/3.4.12/usr/bin/astyle
 MIN_ASTYLE_VER="3.1"
```
Comment (matt335672) [inline scripts/run_astyle.sh:11]: We should probably allow the version to be passed in to the script (like cppcheck) so we only have to set it in one place, but this is fine for now.

File: scripts/install_astyle.sh [hunk 2]
```diff
@@ -80,9 +80,12 @@ fi
...
     FILESDIR=$INSTALL_ROOT/$ASTYLE_VER
 
-    svn checkout ${REPO_URL}/${ASTYLE_VER}/AStyle "$workdir"
-
+    git clone --sparse ${REPO_URL} "$workdir"
     cd "$workdir"
+    git sparse-checkout set AStyle
+    git checkout tags/${ASTYLE_VER}
+
+    cd "AStyle"
```
Comment (matt335672) [inline scripts/install_astyle.sh:85]: Happy with this, but install_cppcheck.sh does it a different way (from https://stackoverflow.com/questions/791959) :-

```
git clone -b "$ASTYLE_VER" --depth 1 "$REPO_URL" "$workdir"
```

**Review by matt335672 — COMMENTED** [review]
A couple of minor comments.

**Comment by matt335672** [comment]
Great news that astyle is being continued. It looks like the code changes are just fixing a few strange things that astyle did anyway, so I'm happy with that.

File: scripts/install_astyle.sh (hunk 2 above)
Comment (metalefty) [reply scripts/install_astyle.sh:85]: It looks simpler! I like it.

File: scripts/run_astyle.sh (hunk 1 above)
Comment (metalefty) [reply scripts/run_astyle.sh:11]: Let's address it later.

**Comment by metalefty** [comment]
Here's the astyle check log with cache deleted.
https://github.com/metalefty/xrdp/actions/runs/7829461430/job/21361981667

**Pull request merged**